pkg/syncer/k8soperator/k8scontroller/namespace/namespace_controller.go

```diff
@@ -0,0 +1,285 @@
+		backOffDurationMapMutex.Lock()
+		var timeout time.Duration
+		if _, exists := backOffDuration[instance.Name]; !exists {
+			backOffDuration[instance.Name] = time.Second
+		}
+		timeout = backOffDuration[instance.Name]
+		backOffDurationMapMutex.Unlock()
+		log.Infof("Reconciling Namespace with Request.Name: %q timeout %q seconds",
+			request.Name, timeout)
+		// If namespace does not have deletion timestamp, then return
+		if instance.DeletionTimestamp.IsZero() {
+			return reconcile.Result{}, "", nil
+		}
+		// Iterate over all PVCs in deleting namespace and remove the CNS finalizer
+		// "cns.vmware.com/pvc-delete-protection" from them, if found
+		var pvcList v1.PersistentVolumeClaimList
+		err = r.client.List(internalCtx, &pvcList, &client.ListOptions{Namespace: instance.Name})
+		if err != nil {
+			msg := fmt.Sprintf("Failed to list PVCs in namespace: %q. Err: %+v",
+				instance.Name, err)
+			recordEvent(internalCtx, r, instance, v1.EventTypeWarning, msg)
+			return reconcile.Result{RequeueAfter: timeout}, csifault.CSIApiServerOperationFault, nil
+		}
+		k8sClient, err := k8s.NewClient(ctx)
+		if err != nil {
+			log.Errorf("Failed to get kubernetes client. Err: %+v", err)
+			return reconcile.Result{}, csifault.CSIInternalFault, err
+		}
+		for _, pvc := range pvcList.Items {
+			log.Infof("Processing pvc %s in namespace %q timeout %q seconds",
+				pvc.Name, pvc.Namespace, timeout)
+			syncer.RemoveCNSFinalizerFromPVCIfTKGClusterDeleted(ctx, k8sClient, &pvc,
+				cnsoperatortypes.CNSVolumeFinalizer, true)
+		}
+		// Iterate over all VolumeSnapshots in deleting namespace and remove the CNS finalizer
+		// "cns.vmware.com/volumesnapshot-protection" from them, if found
+		var vsList snapv1.VolumeSnapshotList
+		err = r.client.List(internalCtx, &vsList, &client.ListOptions{Namespace: instance.Name})
+		if err != nil {
+			msg := fmt.Sprintf("Failed to list VolumeSnapshots in namespace: %q. Err: %+v",
+				instance.Name, err)
+			recordEvent(internalCtx, r, instance, v1.EventTypeWarning, msg)
+			return reconcile.Result{RequeueAfter: timeout}, csifault.CSIApiServerOperationFault, nil
+		}
+		snapshotterClient, err := k8s.NewSnapshotterClient(ctx)
+		if err != nil {
+			log.Errorf("Failed to get snapshotterClient. Err: %v", err)
+			return reconcile.Result{}, csifault.CSIInternalFault, err
+		}
+		for _, vs := range vsList.Items {
+			log.Infof("Processing snapshot %s in namespace %q timeout %q seconds",
+				vs.Name, vs.Namespace, timeout)
+			syncer.RemoveCNSFinalizerFromSnapIfTKGClusterDeleted(ctx, snapshotterClient, &vs,
+				cnsoperatortypes.CNSSnapshotFinalizer, true)
+		}
+		// Cleanup instance entry from backOffDuration map.
+		backOffDurationMapMutex.Lock()
+		delete(backOffDuration, instance.Name)
+		backOffDurationMapMutex.Unlock()
+		log.Infof("Finished Reconcile for Namespace request: %q", request.NamespacedName)
+		return reconcile.Result{}, "", nil
+	}
+	resp, faulttype, err := reconcileNamespaceInternal(ctx)
```

Comment:
can you remove reconcileNamespaceInternal? it seems not necessary.

Reply:
done, removed